src/components/MissionAPP/MissionShelf.jsx

```diff
@@ -0,0 +1,49 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { getMissions, selectMission } from 'features/missions/missionsSlice';
+import { nanoid } from '@reduxjs/toolkit';
+
+const MissionShelf = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getMissions());
+  }, []);
+
+  const { missions, isLoading, error } = useSelector(selectMission);
+
+  const nobooks = (
+    <h3 className="center">No Books Found</h3>
+  );
+
+  if (isLoading) return <h3 className="center">Loading...</h3>;
+  if (error) {
+    return (
+      <h3 className="center">
+        Error:
+        {' '}
+        {error}
+      </h3>
+    );
+  }
+  if (missions.length < 1) return <>{nobooks}</>;
+  const missionArr = missions.map((mission) => (
+    <li key={nanoid()}>
```

Comment:
[OPTIONAL]
- [ ] I suggest using `missionId` as the `key` for the` li `element since It can be helpful if we face some bugs, for example, if two different missions have the same id.

Reply:
Actually, key property in React is just for internal use of the framework. The most unique it is the better.